### PR TITLE
Create crud

### PR DESCRIPTION
**Pull request by jdhiguita**
## Summary
- CRUD methods to Book
## Trello Card
https://trello.com/c/qabJROrE

### CODE REVIEW

File: src/main/java/wolox/training/controllers/BookController.java [hunk 1]
```diff
@@ -1,17 +1,82 @@
...
     }
 
+    /**
+     * Services that find a book according to the author entered by parameter
+     * @param author
+     * @return Book object
+     */
+    @GetMapping("/author/{author}")
+    public Book findByAuthor(@PathVariable String author){
+        return bookRepository.findByAuthor(author).orElseThrow(BookNotFoundException::new);
```
Comment (cramallo) [inline src/main/java/wolox/training/controllers/BookController.java]: You could move this code to a service class, in this case "BookService" class. It's a good practice to call services from the controller, and call the repositories from the services. Check this post https://medium.com/karibu-blog/c%C3%B3mo-aplicar-jpa-en-spring-boot-e45ac5bee3b4

File: src/main/java/wolox/training/exceptions/BookNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package wolox.training.exceptions;
+
+public class BookNotFoundException extends RuntimeException{
+
+    public BookNotFoundException(){
```
Comment (cramallo) [inline src/main/java/wolox/training/exceptions/BookNotFoundException.java:5]: it's ok. Here you could recieve for example a String message paramater to represent a custom message to have an explanation more representative for your application.

File: src/main/java/wolox/training/controllers/BookController.java (hunk 1 above)
Comment (jdhiguita) [reply src/main/java/wolox/training/controllers/BookController.java]: Ok, I create the package and classes. I'm following the instructions on the card and I think that is my mistake

File: src/main/java/wolox/training/exceptions/BookNotFoundException.java (hunk 2 above)
Comment (jdhiguita) [reply src/main/java/wolox/training/exceptions/BookNotFoundException.java:5]: Add other constructors to the class.

**Review by cramallo — APPROVED** [review]

**Review by Gianni-Nicolas — APPROVED** [review]

**Pull request merged**